core/src/test/java/org/apache/iceberg/rest/requests/TestUpdateNamespacePropertiesRequest.java

```diff
@@ -0,0 +1,162 @@
+
+    // Only updates
+    String emptyRemoval = "{\"removals\":[],\"updates\":{\"owner\":\"Hank\"}}";
+    assertRoundTripSerializesEquallyFrom(
+        emptyRemoval, UpdateNamespacePropertiesRequest.builder().updateAll(UPDATES).removeAll(EMPTY_REMOVALS).build());
+
+    assertRoundTripSerializesEquallyFrom(
+        emptyRemoval, UpdateNamespacePropertiesRequest.builder().update("owner", "Hank").build());
+
+    // Only removals
+    String emptyUpdates = "{\"removals\":[\"foo\",\"bar\"],\"updates\":{}}";
+    assertRoundTripSerializesEquallyFrom(
+        emptyUpdates, UpdateNamespacePropertiesRequest.builder().removeAll(REMOVALS).updateAll(EMPTY_UPDATES).build());
+
+    assertRoundTripSerializesEquallyFrom(
+        emptyUpdates, UpdateNamespacePropertiesRequest.builder().remove("foo").remove("bar").build());
+  }
+
+  @Test
+  // Test cases that can't be constructed with our Builder class e2e but that will parse correctly
+  public void testCanDeserializeWithoutDefaultValues() throws JsonProcessingException {
+    // `removals` is left empty.
+    UpdateNamespacePropertiesRequest noRemovals = UpdateNamespacePropertiesRequest.builder().updateAll(UPDATES).build();
+    String jsonWithNullRemovals = "{\"removals\":null,\"updates\":{\"owner\":\"Hank\"}}";
+    UpdateNamespacePropertiesRequest parsed = deserialize(jsonWithNullRemovals);
+    assertEquals(parsed, noRemovals);
+
+    // `removals` is missing from the JSON.
+    String jsonWithMissingRemovals = "{\"updates\":{\"owner\":\"Hank\"}}";
+    assertEquals(deserialize(jsonWithMissingRemovals), noRemovals);
+
+    UpdateNamespacePropertiesRequest noUpdates = UpdateNamespacePropertiesRequest.builder().removeAll(REMOVALS).build();
+    String jsonWithNullUpdates = "{\"removals\":[\"foo\",\"bar\"],\"updates\":null}";
+    assertEquals(deserialize(jsonWithNullUpdates), noUpdates);
+  }
+
+  @Test
+  public void testParseInvalidJson() {
+    // Invalid top-level types
+    String jsonInvalidTypeOnRemovalField =
+        "{\"removals\":{\"foo\":\"bar\"},\"updates\":{\"owner\":\"Hank\"}}";
+    AssertHelpers.assertThrows(
+        "A JSON request with an invalid type for one of the fields should fail to parse",
+        JsonProcessingException.class,
+        () -> deserialize(jsonInvalidTypeOnRemovalField).validate()
+    );
+
+    String jsonInvalidTypeOnUpdatesField =
+        "{\"removals\":[\"foo\":\"bar\"],\"updates\":[\"owner\"]}";
+    AssertHelpers.assertThrows(
+        "A JSON value with an invalid type for one of the fields should fail to parse",
+        JsonProcessingException.class,
+        () -> deserialize(jsonInvalidTypeOnUpdatesField).validate()
+    );
+
+    // Valid top-level (array) types, but at least one entry in the list is not the expected type
+    // NOTE: non-string values that are integral types will still parse into a string list.
+    //    e.g. { removals: [ "foo", "bar", 1234 ] } will parse correctly.
```

Comment:
Just integral types? I don't really think this matters but I'm curious.

Reply:
So far I've tested floats and ints (so JSON number types). Maps / arrays in the JSON will throw, but I'm guessing that Jackson is just calling `.asText()` or something on others.